supertokens_python/recipe/dashboard/recipe_implementation.py

```diff
@@ -33,12 +34,21 @@ async def should_allow_access(
         config: DashboardConfig,
         user_context: Dict[str, Any],
     ) -> bool:
-        api_key_header_value = request.get_header("authorization")
-
-        # We receive the api key as `Bearer API_KEY`, this retrieves just the key
-        api_key = api_key_header_value.split(" ")[1] if api_key_header_value else None
-
-        if api_key is None:
-            return False
-
-        return api_key == config.api_key
+        if not config.api_key:
+            authHeaderValue = request.get_header("authorization")
+
+            if not authHeaderValue:
+                return False
+
+            authHeaderValue = authHeaderValue.split()[1]
+            sessionVerificationResponse = (
+                await Querier.get_instance().send_post_request(
```

Comment:
Python vars should be in snake case. So you should try `auth_header_value`, `session_verification_response`, etc..